capa/features/extractors/common.py

```diff
@@ -0,0 +1,78 @@
+import io
+import logging
+import binascii
+import contextlib
+
+import pefile
+
+import capa.features.extractors.elf
+import capa.features.extractors.pefile
+from capa.features.common import OS, FORMAT_PE, FORMAT_ELF, OS_WINDOWS, Arch, Format
+
+logger = logging.getLogger(__name__)
+
+
+def extract_format(buf):
+    if buf.startswith(b"MZ"):
+        yield Format(FORMAT_PE), 0x0
+    elif buf.startswith(b"\x7fELF"):
+        yield Format(FORMAT_ELF), 0x0
+    else:
+        # we likely end up here:
+        #  1. handling a file format (e.g. macho)
+        #
+        # for (1), this logic will need to be updated as the format is implemented.
+        logger.debug("unsupported file format: %s", binascii.hexlify(buf[:4]).decode("ascii"))
+        return
+
+
+def extract_arch(buf):
+    if buf.startswith(b"MZ"):
+        yield from capa.features.extractors.pefile.extract_file_arch(pefile.PE(data=buf), "hack: path not provided")
```

Comment:
is `hack: path not provided` supposed to stay here like this?

Reply:
yeah... but let me add some documentation inline.

this routine accepts two parameters: pe and file path. but the second is not used. its provided just for consistency with other file-scope extractors. maybe it makes sense to make it a kwarg instead.

Reply:
fc73787